pkg/codegen/go/gen_program.go

```diff
@@ -510,8 +512,11 @@ func (g *generator) getPulumiImport(pkg, vPath, mod, name string) string {
 	modSplit := strings.Split(mod, "/")
 	// account for mods like "eks/ClusterVpcConfig" index...
 	if len(modSplit) > 1 {
-		if modSplit[0] == "" || modSplit[0] == "index" {
+		if modSplit[0] == "" || modSplit[0] == IndexToken {
 			imp = fmt.Sprintf("github.com/pulumi/pulumi-%s/sdk%s/go/%s", pkg, vPath, pkg)
+			if info.ImportBasePath != "" {
+				imp = info.ImportBasePath
```

Comment:
@iwahbe Friel and I decided that we should not alias imports unless there is an explicit alias mapping in the schema, so this should be `thirdparty.NewProvider`. There was an inconsistency where we here we should be importing the `ImportBasePath` if `mod == "" || mod == "index"` (so the import would be `.../thirdparty`), but then before calling `getModOrAlias`, setting `mod = pkg` on line 625 which results in `modOrAlias` being `other`.